tests/test_plottingfuncs.py

```diff
@@ -1,53 +1,129 @@
 """Tests of plotting functions."""
 import pytest
+from pathlib import Path
 import numpy as np
+import skimage.io as io
 
 from topostats.filters import Filters
 from topostats.grains import Grains
-from topostats.plottingfuncs import plot_and_save
+from topostats.plottingfuncs import Images
+
+@pytest.mark.parametrize("data2, axes_colorbar, region_properties", [
+(np.random.rand(10,10), True, None),
+(None, True, None),
+(None, False, True)
+])
+def test_save_figure(
+    data2: np.ndarray,
+    axes_colorbar: bool,
+    region_properties: bool,
+    minicircle_grain_region_properties_post_removal: Grains,
+    tmp_path: Path):
+    """Tests that an image is saved and a figure returned"""
+    if region_properties:
+        region_properties=minicircle_grain_region_properties_post_removal
+    fig, ax = Images(
+        data=np.random.rand(10,10),
+        output_dir=tmp_path,
+        filename="result.png",
+        data2=data2,
+        colorbar=axes_colorbar,
+        axes=axes_colorbar,
+        region_properties=region_properties,
+    ).save_figure()
+    assert Path(tmp_path / "result.png").exists()
+    assert fig is not None
+    assert ax is not None
+
+
+def test_save_array_figure(tmp_path: Path):
+    """Tests that the image array is saved"""
+    Images(
+        data=np.random.rand(10,10),
```

Comment:
The idea here was that the method linked to this is test basically just a nice wrapper for `plt.imsave` so all that would be tested with a mpl compare its that its arguments work and this runs, hence why I thought the `pytest-mpl` might have been unnecessary as this would test an image that is already tested elsewhere (in `test_plot_and_save_no_axes_no_colorbar`). 

I'll add the `image_random` to make sure its consistent but if you disagree with the need for the mpl-compare, let me know and I'll add it in

Reply:
No, not at all, as I say I thought the suggestion might be overkill, and even reusing the `image_random` is to some extent excessive. I just clocked it and knew that the fixture existed.